src/device-state.ts

```diff
@@ -684,32 +684,37 @@ export function reportCurrentState(newState: DeviceReport = {}) {
 	emitAsync('change', undefined);
 }
 
-export async function reboot(force?: boolean, skipLock?: boolean) {
-	await updateLock.abortIfHUPInProgress({ force });
-	await applicationManager.stopAll({ force, skipLock });
-	logger.logSystemMessage('Rebooting', {}, 'Reboot');
-	const $reboot = await dbus.reboot();
-	shuttingDown = true;
-	emitAsync('shutdown', undefined);
-	return await $reboot;
+export async function reboot(force: boolean) {
+	// Multi-app warning!
+	const appId = (await config.get('applicationId')) as number;
+	return updateLock.lock(appId, { force }, async () => {
+		logger.logSystemMessage('Rebooting', {}, 'Reboot');
+		const $reboot = await dbus.reboot();
+		shuttingDown = true;
+		emitAsync('shutdown', undefined);
+		return $reboot;
+	});
 }
 
-export async function shutdown(force?: boolean, skipLock?: boolean) {
+export async function shutdown(force: boolean) {
 	await updateLock.abortIfHUPInProgress({ force });
-	await applicationManager.stopAll({ force, skipLock });
-	logger.logSystemMessage('Shutting down', {}, 'Shutdown');
-	const $shutdown = await dbus.shutdown();
-	shuttingDown = true;
-	emitAsync('shutdown', undefined);
-	return $shutdown;
+	// Multi-app warning!
+	const appId = (await config.get('applicationId')) as number;
```

Comment:
get application IDs from the database and ensure you can create a lock in each of these apps.